src/main/webapp/country.js

```diff
@@ -0,0 +1,103 @@
+  const cities = await citiesResponse.json();
+
+  const citiesInCountry = [];
+  for (const city in cities) {
+    if (city.countryID === id) {
+      citiesInCountry.push(city);
+    }
+  }
+
+  displayCityResults(citiesInCountry, country);
+}
+
+document.addEventListener('DOMContentLoaded', (event) => {
+  loading();
+});
+
+/**
+ * Loads seach paramerter.
+ */
+function loading() {
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const query = urlParams.get('id');
+  searching(query);
+}
+
+/**
+ * @param {citiesInCountry} citiesInCountry
+ * @param {country} country
+ */
+function displayCityResults(citiesInCountry, country) {
+  // add name of country as header element
+  const countryElement = document.createElement('H1');
+  countryElement.innerText = country.name;
+  document.getElementById('country-name').appendChild(countryElement);
+
+  // add do's and dont's as two seperate unordered lists
+  const doElement = document.createElement('ul');
+  for (const doTip in country.cultureDos) {
+    if (doTip != null) {
+      const doItem = document.createElement('li');
+      doItem.appendChild(doTip);
+      doElement.appendChild(doItem);
+    }
+  }
+  document.getElementById('dos').appendChild(doElement);
+
+  const dontElement = document.createElement('ul');
+  for (const dontTip in country.cultureDonts) {
+    if (dontTip != null) {
+      const dontItem = document.createElement('li');
+      dontItem.appendChild(dontTip);
+      dontElement.appendChild(dontItem);
+    }
+  }
+  document.getElementById('donts').appendChild(dontElement);
+
+  // add language spoken to "dos" list
+  const languageElement = document.createElement('li');
+  languageElement.appendChild('Learn a few words in ' + country.languages);
+  document.getElementById('language').appendChild(languageElement);
+
+  // add table with cities as individual rows
+  const citiesTable = document.getElementById('citiesTable');
+  for (const city in citiesInCountry) {
+    if (city != null) {
+      const row = citiesTable.insertRow(-1);
+
+      const cell1 = row.insertCell(0);
+      const cell2 = row.insertCell(1);
+
+      cell1.innerHTML = city.name;
+      cell2.innerHTML = city.description;
+
+      // add table with events from various cities as individual rows
+      const eventsTable = document.getElementById('eventsTable');
+      for (const event in city.events) {
+        if (event != null) {
+          const row = eventsTable.insertRow(-1);
+
+          const cell1 = row.insertCell(0);
+          const cell2 = row.insertCell(1);
+          const cell3 = row.insertCell(2);
+          const cell4 = row.insertCell(3);
+          const cell5 = row.insertCell(4);
+
+          cell1.innerHTML = event.name;
+          cell2.innerHTML = event.description;
+          cell3.innerHTML = event.date;
+          cell4.innerHTML = event.location;
+          cell5.innerHTML = event.pricing;
+        }
+      }
```

Comment:
I might suggest instead of adding events to the country page, maybe add a [link](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a)(using `href`) or a [button](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button)(using `onclick`) in the city row that takes the user to the city page for that city? If a link, it could just go around the name of the city

Reply:
FYI, I merged the PR to get the changes in, but if you like this suggestion or parts of it please feel free to address in a new PR